internal/cmdrender/runtime/container.go

```diff
@@ -103,3 +126,34 @@ func (f *ContainerFn) getDockerCmd() (*exec.Cmd, context.CancelFunc) {
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	return exec.CommandContext(ctx, path, args...), cancel
 }
+
+// prepareImage will check local images and pull it if it doesn't
+// exist.
+func (f *ContainerFn) prepareImage() error {
+	// check image existence
+	path := "docker"
+	args := []string{"image", "ls", f.Image}
+	cmd := exec.Command(path, args...)
+	var output []byte
+	var err error
+	if output, err = cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("failed to check function %q: %w", f.Image, err)
```

Comment:
go/kpt-errors-style-guide

what does "failed to check function" mean to user? The noun here is "function image". Audit all error messages against the style guide.

Reply:
I want to avoid implementation details like "image". But will update it.

Reply:
"image" is not implementation. It also appears in our interfaces in the Kptfile and explained in docs. But think about what you're telling the user:

`failed to check function: <something very docker image specific>`

Reply:
updated